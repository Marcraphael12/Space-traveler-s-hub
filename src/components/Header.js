import Logo from '../images/planet.png';
import { NavLink } from 'react-router-dom';

const Header = () => (
  <header className="header">
    <div className="logo">
      <img src={Logo} alt="logo" />
      <span>Space Travelers' Hub</span>
    </div>
    <nav>
      <ul className="nav-items">
        <li>
          <NavLink to="/">Rockets</NavLink>
        </li>
        <li>
          <NavLink to="missions">Missions</NavLink>
        </li>
      </ul>
    </nav>
  </header>
);

export default Header;
