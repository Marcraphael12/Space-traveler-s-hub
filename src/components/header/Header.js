import './header.scss';
import Logo from '../..//images/planet.png';
import { NavLink } from 'react-router-dom';

const Header = () => (
  <header className="header container">
    <div className="header__logo">
      <img className="header__logo-img" src={Logo} alt="logo" />
      <span className="header__logo-title">Space Travelers' Hub</span>
    </div>
    <nav>
      <ul className="header__nav-items">
        <li className="header__nav-item">
          <NavLink className="header__nav-link" to="/">
            Rockets
          </NavLink>
        </li>
        <li className="header__nav-item header__nav-item--br">
          <NavLink className="header__nav-link" to="missions">
            Missions
          </NavLink>
        </li>
        <li className="header__nav-item">
          <NavLink className="header__nav-link" to="my-profile">
            My Profile
          </NavLink>
        </li>
      </ul>
    </nav>
  </header>
);

export default Header;
