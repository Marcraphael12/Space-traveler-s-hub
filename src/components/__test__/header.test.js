import TestRenderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';
import {
  BrowserRouter as Router,
} from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../../redux/configureStore';
import Header from '../header/Header';

describe('Render avigationr', () => {
  it('Displays correctly the header with the navigation bar', () => {
    const header = TestRenderer.create(
      <Provider store={store}>
        <Router>
          <Header />
        </Router>
      </Provider>,
    ).toJSON();
    expect(header).toMatchSnapshot();
  });
});