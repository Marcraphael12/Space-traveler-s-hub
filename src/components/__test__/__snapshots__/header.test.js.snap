// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render avigationr Displays correctly the header with the navigation bar 1`] = `
<header
  className="header container"
>
  <div
    className="header__logo"
  >
    <img
      alt="logo"
      className="header__logo-img"
      src="planet.png"
    />
    <span
      className="header__logo-title"
    >
      Space Travelers' Hub
    </span>
  </div>
  <nav>
    <ul
      className="header__nav-items"
    >
      <li
        className="header__nav-item"
      >
        <a
          aria-current="page"
          className="header__nav-link active"
          href="/"
          onClick={[Function]}
        >
          Rockets
        </a>
      </li>
      <li
        className="header__nav-item header__nav-item--br"
      >
        <a
          className="header__nav-link"
          href="/missions"
          onClick={[Function]}
        >
          Missions
        </a>
      </li>
      <li
        className="header__nav-item"
      >
        <a
          className="header__nav-link"
          href="/my-profile"
          onClick={[Function]}
        >
          My Profile
        </a>
      </li>
    </ul>
  </nav>
</header>
`;
